modules/tensor_mechanics/src/actions/LineElementAction.C

```diff
@@ -0,0 +1,827 @@
+      "nodal_Iyy", "nodal_Iyy>=0.0", "Nodal moment of inertia in the y direction.");
+  params.addRangeCheckedParam<Real>(
+      "nodal_Izz", "nodal_Izz>=0.0", "Nodal moment of inertia in the z direction.");
+  params.addParam<Real>("nodal_Ixy", 0.0, "Nodal moment of inertia in the xy direction.");
+  params.addParam<Real>("nodal_Ixz", 0.0, "Nodal moment of inertia in the xz direction.");
+  params.addParam<Real>("nodal_Iyz", 0.0, "Nodal moment of inertia in the yz direction.");
+  params.addParam<RealGradient>(
+      "nodal_x_orientation",
+      "Unit vector along the x direction if different from global x direction.");
+  params.addParam<RealGradient>(
+      "nodal_y_orientation",
+      "Unit vector along the y direction if different from global y direction.");
+  params.addParam<std::vector<BoundaryName>>(
+      "boundary",
+      "The list of boundary IDs from the mesh where the nodal "
+      "mass/inertia will be applied.");
+  return params;
+}
+
+LineElementAction::LineElementAction(const InputParameters & params)
+  : Action(params),
+    _rotations(0),
+    _velocities(0),
+    _accelerations(0),
+    _rot_velocities(0),
+    _rot_accelerations(0),
+    _save_in(getParam<std::vector<AuxVariableName>>("save_in")),
+    _diag_save_in(getParam<std::vector<AuxVariableName>>("diag_save_in")),
+    _subdomain_names(getParam<std::vector<SubdomainName>>("block")),
+    _subdomain_ids(),
+    _strain(getParam<MooseEnum>("strain").getEnum<Strain>()),
+    _dynamic_consistent_inertia(getParam<bool>("dynamic_consistent_inertia")),
+    _dynamic_nodal_translational_inertia(getParam<bool>("dynamic_nodal_translational_inertia")),
+    _dynamic_nodal_rotational_inertia(getParam<bool>("dynamic_nodal_rotational_inertia")),
+    _truss(getParam<bool>("truss"))
+{
+  // check if a container block with common parameters is found
+  auto action = _awh.getActions<CommonLineElementAction>();
+  if (action.size() == 1)
+    _pars.applyParameters(action[0]->parameters());
+
+  if (!isParamValid("displacements"))
+    mooseError("LineElementAction: A vector of displacement variable names should be provided as "
+               "input using `displacements`.");
+
+  _displacements = getParam<std::vector<VariableName>>("displacements");
+  _ndisp = _displacements.size();
+
+  // determine if displaced mesh is to be used
+  _use_displaced_mesh = (_strain != Strain::SMALL_STRAIN_AND_ROTATION);
+  if (params.isParamSetByUser("use_displaced_mesh"))
+  {
+    bool use_displaced_mesh_param = getParam<bool>("use_displaced_mesh");
+    if (use_displaced_mesh_param != _use_displaced_mesh && params.isParamSetByUser("strain"))
+      mooseError("LineElementAction: Wrong combination of use displaced mesh and strain model.");
+    _use_displaced_mesh = use_displaced_mesh_param;
+  }
+
+  if (_save_in.size() != 0 && _save_in.size() != _ndisp)
+    mooseError("LineElementAction: Number of save_in variables should equal to the number of "
```

Comment:
General statement: Use `paramError` for all parameter related errors. It just takes the name of the parameter as the first argument and the error message as the second argument. It'll provide line numbers into the input file to assist users in finding the problem.